guvnor-m2repo-editor/guvnor-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/helpers/HttpGetHelper.java

```diff
@@ -46,35 +46,37 @@
     @Inject
     private GuvnorM2Repository repository;
 
-    public void handle( final HttpServletRequest request,
-                        final HttpServletResponse response,
-                        final ServletContext context ) throws IOException {
+    public void handle(final HttpServletRequest request,
+                       final HttpServletResponse response,
+                       final ServletContext context) throws IOException {
         String requestedFile = request.getPathInfo();
 
-        if ( requestedFile == null ) {
-            response.sendError( HttpServletResponse.SC_NOT_FOUND );
+        if (requestedFile == null) {
+            response.sendError(HttpServletResponse.SC_NOT_FOUND);
             return;
         }
 
-        requestedFile = URLDecoder.decode( requestedFile,
-                                           "UTF-8" );
+        requestedFile = URLDecoder.decode(requestedFile,
+                                          "UTF-8");
+
+        String repositoryName = request.getParameter("repository");
```

Comment:
@adrielparedes You may be able to guarantee this when we attempt to download a JAR from the ```M2RepoEditorPresenter``` screen; however I was meaning what if the workbench repository is added to a project's ```pom.xml``` and Maven (from the CLI) tries to download JARs; or someone is browsing the Workbench's Maven repository in a browser; etc. In these scenarios the ```HttpServletRequest``` is not controlled by *us* and hence may not have a ```repository``` parameter... or am I still missing something?